data/query_test.go

```diff
@@ -0,0 +1,206 @@
+package data
+
+import (
+	"context"
+	"net/url"
+	"testing"
+
+	"github.com/ONSdigital/dp-frontend-search-controller/config"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestUnitReviewQuerySuccess(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	Convey("Given valid url query", t, func() {
+		cfg, err := config.Get()
+		So(err, ShouldBeNil)
+
+		urlQuery := url.Values{
+			"q":      []string{"housing"},
+			"filter": []string{"article"},
+			"sort":   []string{"relevance"},
+			"limit":  []string{"10"},
+			"page":   []string{"1"},
+		}
+
+		Convey("When ReviewQuery is called", func() {
+			validatedQueryParams, err := ReviewQuery(ctx, cfg, urlQuery)
+
+			Convey("Then successfully review and return validated query parameters", func() {
+				So(validatedQueryParams, ShouldResemble, SearchURLParams{
+					Query: "housing",
+					Filter: Filter{
+						Query:           []string{"article"},
+						LocaliseKeyName: []string{"Article"},
+					},
+					Sort: Sort{
+						Query:           "relevance",
+						LocaliseKeyName: "Relevance",
+					},
+					Limit:       10,
+					CurrentPage: 1,
+				})
+			})
+
+			Convey("And return no errors", func() {
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+}
+
+func TestUnitReviewQueryFailure(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	Convey("Given failure to review pagination due to invalid pagination parameters provided", t, func() {
+		cfg, err := config.Get()
+		So(err, ShouldBeNil)
+
+		urlQuery := url.Values{
+			"q":      []string{"housing"},
+			"filter": []string{"article"},
+			"sort":   []string{"relevance"},
+			"limit":  []string{"10"},
+			"page":   []string{"10000000"},
+		}
+
+		Convey("When ReviewQuery is called", func() {
+			_, err := ReviewQuery(ctx, cfg, urlQuery)
+
+			Convey("Then return an error", func() {
+				So(err, ShouldNotBeNil)
+			})
+		})
+	})
+
+	Convey("Given failure to review filter due to invalid filters provided", t, func() {
+		cfg, err := config.Get()
+		So(err, ShouldBeNil)
+
+		urlQuery := url.Values{
+			"q":      []string{"housing"},
+			"filter": []string{"INVALID"},
+			"sort":   []string{"relevance"},
+			"limit":  []string{"10"},
+			"page":   []string{"1"},
+		}
+
+		Convey("When ReviewQuery is called", func() {
+			_, err := ReviewQuery(ctx, cfg, urlQuery)
+
+			Convey("Then return an error", func() {
+				So(err, ShouldNotBeNil)
+			})
+		})
```

Comment:
If the query is `empty`, it will retrieve `all` the search results available (up to a maximum of `500` results)
If the query is `invalid`, the `search-api` would retrieve `0` search results